The flag is not recognized by GCC, so just remove it.

Author: Alexander Grund (TU Dresden)
diff --git a/third_party/xla/xla/codegen/intrinsic/cpp/cc_to_llvm_ir.bzl b/third_party/xla/xla/codegen/intrinsic/cpp/cc_to_llvm_ir.bzl
--- a/third_party/xla/xla/codegen/intrinsic/cpp/cc_to_llvm_ir.bzl
+++ b/third_party/xla/xla/codegen/intrinsic/cpp/cc_to_llvm_ir.bzl
@@ -106,7 +106,6 @@ def cc_ir_header(name, src, deps = [], copts = [], **kwargs):
         "-fno-builtin",
         "-Wno-psabi",
         "-std=c++17",
-        "-fno-experimental-sanitize-metadata=all",
         "-fno-sanitize=all",
     ] + copts
 
